crypto/Connect.cpp c++11 decltype

commit https://github.com/open-eid/libdigidocpp/commit/10287ae2a1c07b56d72a308a523c33584b365eda#diff-b96d913077879e2814ffaccd2d949913 changed the decltype but maybe missed ? some https://github.com/open-eid/libdigidocpp/blob/15a83ba88e6292fff76d90e3da6e4944448012a0/src/crypto/Connect.cpp#L98 and this fails with gcc 7.1.1

`[ 34%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/log.cpp.o
[ 35%] Building CXX object src/CMakeFiles/digidocpp_priv.dir/crypto/Connect.cpp.o
/home/edr/libdigidocpp/src/src/crypto/Connect.cpp: In constructor ‘digidoc::Connect::Connect(const string&, const string&, int, const string&, const digidoc::X509Cert&)’:
/home/edr/libdigidocpp/src/src/crypto/Connect.cpp:60:28: error: ‘strlen’ was not declared in this scope
     string url = (_path && strlen(_path) == 1 && _path[0] == '/' && _url[_url.size() - 1] != '/') ? _url + "/" : _url;
                            ^~~~~~
/home/edr/libdigidocpp/src/src/crypto/Connect.cpp:60:28: note: suggested alternative: ‘mbrlen’
     string url = (_path && strlen(_path) == 1 && _path[0] == '/' && _url[_url.size() - 1] != '/') ? _url + "/" : _url;
                            ^~~~~~
                            mbrlen
/home/edr/libdigidocpp/src/src/crypto/Connect.cpp:98:56: error: ‘function’ was not declared in this scope
         ssl.reset(SSL_CTX_new(SSLv23_client_method()), function<void(SSL_CTX*)>(SSL_CTX_free));
                                                        ^~~~~~~~
/home/edr/libdigidocpp/src/src/crypto/Connect.cpp:98:56: note: suggested alternative: ‘union’
         ssl.reset(SSL_CTX_new(SSLv23_client_method()), function<void(SSL_CTX*)>(SSL_CTX_free));
                                                        ^~~~~~~~
                                                        union
/home/edr/libdigidocpp/src/src/crypto/Connect.cpp:98:65: error: expected primary-expression before ‘void’
         ssl.reset(SSL_CTX_new(SSLv23_client_method()), function<void(SSL_CTX*)>(SSL_CTX_free));
                                                                 ^~~~
make[2]: *** [src/CMakeFiles/digidocpp_priv.dir/build.make:439: src/CMakeFiles/digidocpp_priv.dir/crypto/Connect.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:92: src/CMakeFiles/digidocpp_priv.dir/all] Error 2
make: *** [Makefile:141: all] Error 2`

--- a/src/crypto/Connect.cpp
+++ b/src/crypto/Connect.cpp
@@ -29,6 +29,8 @@
 
 #include <zlib.h>
 
+#include <cstring>
+#include <functional>
 #include <thread>
 
 #ifdef __ANDROID__

